src/core/border_router/routing_manager.cpp

```diff
@@ -1739,6 +1739,16 @@ void RoutingManager::DiscoveredPrefixTable::Entry::SetFrom(const RouteInfoOption
     mLastUpdateTime          = TimerMilli::GetNow();
 }
 
+void RoutingManager::DiscoveredPrefixTable::Entry::SetFrom(const PrefixTableEntry &aPrefixTableEntry)
+{
+    mPrefix.Set(aPrefixTableEntry.mPrefix.mPrefix.mFields.m8, aPrefixTableEntry.mPrefix.mLength);
```

Comment:
I think this can work and is simpler:
```suggestion
    mPrefix = AsCoreType(&aPrefixTableEntry.mPrefix);
```